tests/testthat/_snaps/ptype-abbr-full.md

```diff
@@ -0,0 +1,24 @@
+# data.frames have good default abbr and full methods
+
+    Code
+      df <- foobar(data.frame(x = 1, y = "", z = TRUE))
+      vec_ptype_abbr(df)
+    Output
+      [1] "vctrs_fb[,3]"
+    Code
+      vec_ptype_full(df)
+    Output
+      [1] "vctrs_foobar<\n  x: double\n  y: character\n  z: logical\n>"
+
+---
+
+    Code
+      dt <- data.table::data.table(x = 1, y = 2, z = 3)
+      vec_ptype_abbr(dt)
+    Output
+      [1] "data.tbl[,3]"
```

Comment:
yup I thought abbreviating unknown data frames was more consistent with the documented behaviour of `vec_ptype_abbr()`, and more useful regarding its intended usage.

Reply:
Should we add a data.table method for the sake of our dt users? I don't imagine the data.table team will be interested in maintaining tidyverse integration in their package, so we could do it here in vctrs. In this case, I suppose we'd abbreviate to `"dt"`?

Reply:
Seems reasonable since we have ptype2/cast methods for them

Maybe pull out a `type-data-table.R` file starting from here?
https://github.com/r-lib/vctrs/blob/e86bb9a4b57459eb56099c0b52af37877a9bf515/R/type-misc.R#L126

I also noticed we have some tests that were skipped because we weren't Suggesting data table until now, maybe we can update that too
https://github.com/r-lib/vctrs/blob/e86bb9a4b57459eb56099c0b52af37877a9bf515/tests/testthat/test-type-misc.R#L33

All of that together might be enough for a separate PR?